olp-cpp-sdk-core/tests/thread/ThreadPoolTaskSchedulerTest.cpp

```diff
@@ -280,3 +280,20 @@ TEST(ThreadPoolTaskSchedulerTest, Move) {
   thread_pool.reset();
   testing::Mock::VerifyAndClearExpectations(&mockop);
 }
+
+TEST(ThreadPoolTaskSchedulerTest, ScheduleTaskViaExecuteOrScheduleFunction) {
+  auto scheduler = std::make_shared<olp::thread::ThreadPoolTaskScheduler>(1);
+
+  std::promise<void> promise;
+  auto future = promise.get_future();
+  olp::thread::ExecuteOrSchedule(scheduler, [&]() { promise.set_value(); });
+
+  EXPECT_EQ(future.wait_for(std::chrono::milliseconds(kMaxWaitMs)),
+            std::future_status::ready);
+}
+
+TEST(ThreadPoolTaskSchedulerTest, ExecuteTaskImmediately) {
+  auto counter(0u);
+  olp::thread::ExecuteOrSchedule(nullptr, [&]() { counter++; });
+  EXPECT_EQ(counter, 1);
+}
```

Comment:
Merge please into one test with SCOPED_TRACE, calling it ExecuteOrSchedule.

Reply:
done